### ISSUE TITLE
Meta-field definitions in a mappings object stops auto inferencing

### ISSUE THREAD

**Issue by siddharthlatest** [appbaseio/gem#4]
Try out live with a `_ttl` meta field [magic link](https://opensource.appbase.io/gem/#?input_state=XQAAAAKOAQAAAAAAAAA9iImmVFabo7XsB6A419CICVNEnslh5QMbF3MIxK79prp7RmFG9z53A-vt-BB1OBWqVdcojsgKXBr__CsazDP9ce1RffqZaBFKbcTCv2ROYZjSNa8GBztQwD9Zclvey5g5QIiS8ZVp2uazIsTdboXjvgJkdxzC5jABi6iKF35W1hF_lU0rW-TEXsW5k4jxQqt5oqQXug1kcTx6bFrLMj-jRq-a7om0D4PKOhW6J-D9Nx6iKLM2WnUndav_qJ794gS1iFtjP__4x-i_).

The mappings object parser should let the meta-fields be intact while providing auto-inference for all the properties object. The current parser behavior stops inferring data types for properties added after the meta field addition.

Besides `_ttl`, there's `_timestamp`, `_all`, `_type` among many other possible meta fields [documented here](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/mapping-fields.html).

**Comment by siddharthlatest** [comment]
Importantly, it also returns a 400 error on trying to apply mappings. Incidentally, the same request works with a direct REST API call so the mappings object definition has no issues.

Error returned by GEM -
![](https://i.imgur.com/HDKYNKU.png)

Successful request via a direct PUT request -
![](https://i.imgur.com/z5UTsW7.png)

**Comment by farhan687** [comment]
For now allow meta fields: https://github.com/appbaseio/gem/blob/dev/app/mappingContainer/importContainer/ImportResult.js#L17 
```js
[
			'_all',
			'_field_names',
			'_timestamp',
			'_ttl'
		]
```

let me know about ui, where to show this after apply mapping.

and for direct PUT request can you show me the request url?

**Comment by farhan687** [comment]
Fixed over here: 7ce3c7eabae4818e4b5c1cb8e27ec83c56e660c3
